crates/database/src/settlement_call_data.rs

```diff
@@ -0,0 +1,60 @@
+use {
+    crate::{auction::AuctionId, PgTransaction},
+    sqlx::PgConnection,
+    std::ops::DerefMut,
+};
+
+#[derive(Debug, Clone, PartialEq, sqlx::FromRow)]
+pub struct SettlementCallData {
+    pub auction_id: AuctionId,
+    pub call_data: Vec<u8>,
+    pub uninternalized_call_data: Vec<u8>,
+}
+
+pub async fn insert(
+    ex: &mut PgTransaction<'_>,
```

Comment:
This should allow you to call this function with an connection returned from `db.acquire()` or a transaction returned from `db.begin()`.

```suggestion
    ex: &mut PgConnection,
```

Reply:
Fixed